@xen-orchestra/xapi/vdi.mjs

```diff
@@ -183,6 +189,9 @@ class Vdi {
       error.nbdClient = nbdClient
       nbdClient?.disconnect()
       exportStream?.destroy()
+      if (taskRef !== undefined) {
+        this.task_destroy(taskRef).catch(warn)
+      }
```

Comment:
This pattern complexify the code, I'm wondering whether we should use `$defer.onFailure()` instead.